packages/next/server/web/sandbox/context.ts

```diff
@@ -153,9 +159,12 @@ async function createModuleContext(options: ModuleContextOptions) {
 
           const key = fn.toString()
           if (instantiatedFromBuffer && !warnedWasmCodegens.has(key)) {
-            const warning = new Error(
-              "Dynamic WASM code generation ('WebAssembly.instantiate' with a buffer parameter) not allowed in Middleware.\n" +
-                'Learn More: https://nextjs.org/docs/messages/middleware-dynamic-wasm-compilation'
+            const warning = getServerError(
+              new Error(
```

Comment:
I'm not sure if we want to decorate this one because the error is going to show in the Logs the location where it originated while in this case we want to simply show the error message.

Reply:
I was also surprised. But the existing test suite expect this warning to be sanitized.
Here is how it looks in logs right now (as it was before):
```shell
warn - (middleware)/lib/wasm.js (22:23) @ usingWebAssemblyCompile
DynamicWasmCodeGenerationWarning: Dynamic WASM code generation (e. g. 'WebAssembly.compile') not allowed in Middleware.
Learn More: https://nextjs.org/docs/messages/middleware-dynamic-wasm-compilation "./middleware"
  20 |
  21 | export async function usingWebAssemblyCompile(x) {
> 22 |   const module = await WebAssembly.compile(SQUARE_WASM_BUFFER)
     |                       ^
  23 |   const instance = await WebAssembly.instantiate(module, {})
  24 |   return { value: instance.exports.square(x) }
  25 | }
```

And here is how it looks if we don't decorate it:
```shell
warn  - DynamicWasmCodeGenerationWarning: Dynamic WASM code generation (e. g. 'WebAssembly.compile') not allowed in Middleware.
Learn More: https://nextjs.org/docs/messages/middleware-dynamic-wasm-compilation "./middleware"
    at usingWebAssemblyCompile (webpack-internal:///(middleware)/./lib/wasm.js:86:26)
    at Object.middleware [as handler] (webpack-internal:///(middleware)/./middleware.js:27:110)
    at adapter (webpack-internal:///(middleware)/../../../packages/next/dist/server/web/adapter.js:57:33)
    at Module.__WEBPACK_DEFAULT_EXPORT__ (webpack-internal:///(middleware)/../../../packages/next/dist/build/webpack/loaders/next-middleware-loader.js?absolutePagePath=%2FUsers%2Fdamien%2Fdev%2Fnext.js%2Ftest%2Fintegration%2Fmiddleware-dynamic-code%2Fmiddleware.js&page=%2Fmiddleware&matcherRegexp=!:24:172)
    at /Users/damien/dev/next.js/packages/next/dist/server/web/sandbox/sandbox.js:34:73
    at async DevServer.runMiddleware (/Users/damien/dev/next.js/packages/next/dist/server/next-server.js:794:26)
    at async DevServer.runMiddleware (/Users/damien/dev/next.js/packages/next/dist/server/dev/next-dev-server.js:482:28)
    at async Object.fn (/Users/damien/dev/next.js/packages/next/dist/server/next-server.js:882:30)
    at async Router.execute (/Users/damien/dev/next.js/packages/next/dist/server/router.js:207:36)
    at async DevServer.run (/Users/damien/dev/next.js/packages/next/dist/server/base-server.js:653:29)
```

Reply:
Very nice